lib/_http_server.js

```diff
@@ -609,7 +613,9 @@ function resOnFinish(req, res, socket, state, server) {
     req._dump();
 
   res.detachSocket(socket);
+  req.closed = true;
   req.emit('close');
+  res.closed = true;
```

Comment:
I had to move `res.closed` into the same tick in order to make other pending fixes work.